spec/models/manageiq/providers/google/network_manager/refresh_parser_spec.rb

```diff
@@ -0,0 +1,186 @@
+require 'digest'
+
+describe ManageIQ::Providers::Google::NetworkManager::RefreshParser do
+  let(:ems) { FactoryGirl.create(:ems_google, :project => "manageiq-dev") }
+  let(:az) { FactoryGirl.create(:availability_zone_google) }
+  let(:vm) do
+    FactoryGirl.create(:vm_google,
+                       :ext_management_system => ems,
+                       :ems_ref               => 123,
+                       :availability_zone     => az)
+  end
+  # These tests use some fairly aggressive stubbing. Specifically, all of the
+  # fog calls are stubbed out. This allows us to test the refresh logic without
+  # resorting to VCR casettes.
```

Comment:
I'd say violent stubbing, to be honest ;) As I'm unfamiliar with how the refresh parsers work I'll need a little more time to glance and see if we can't reduce this a bit more.